packages/core/src/modules/credentials/formats/indy/IndyCredentialFormatService.ts

```diff
@@ -89,7 +90,12 @@ export class IndyCredentialFormatService extends CredentialFormatService {
       throw new AriesFrameworkError('Missing payload in createProposal')
     }
 
-    const attachment: Attachment = this.getFormatData(options.credentialFormats.indy?.payload, formats.attachId)
+    // Use class instance instead of interface, otherwise this causes interoperability problems
+    let proposal = new CredPropose(options.credentialFormats.indy?.payload)
+    proposal = JsonTransformer.toJSON(proposal)
+
+    const attachment: Attachment = this.getFormatData(proposal, formats.attachId)
```

Comment:
In TS, different than most programming languages like e.g. Java, C#, ..., common behaviour is to only use type annotations when you need to. the `getFormatData` method has a return type of `Attachment` making the explicit definition of `Attachment` unnecessary. This in generally also the pattern we use in AFJ.

```suggestion
    const attachment = this.getFormatData(proposal, formats.attachId)
```